integration-tests/tests/integration/toolkit.rs

```diff
@@ -0,0 +1,70 @@
+use madara_prover_rpc_client::services::prover::prover_proto::prover_client::ProverClient;
+use madara_prover_rpc_client::services::starknet_prover::starknet_prover_proto::starknet_prover_client::StarknetProverClient;
+use madara_prover_rpc_server::error::ServerError;
+use madara_prover_rpc_server::{run_grpc_server, BindAddress};
+use rand::distributions::Alphanumeric;
+use rand::Rng;
+use rstest::fixture;
+use std::path::PathBuf;
+use std::time::Duration;
+use tokio::net::UnixStream;
+use tokio::task::JoinHandle;
+use tonic::transport::{Endpoint, Uri};
+use tower::service_fn;
+use test_fixtures::prover_in_path;
+
+pub type RpcServer = JoinHandle<Result<(), ServerError>>;
+
+fn random_string(length: usize) -> String {
+    (0..length)
+        .map(|_| rand::thread_rng().sample(Alphanumeric) as char)
```

Comment:
You could consider using a crypto RNG here if a weak RNG could lead to attacks (stealing someone elses' socket, etc.) Probably not a big deal.